### PR TITLE
add support for txt and pdf in file_handler

### PR DESCRIPTION
**Pull request by dusvyat**
## Description

Add support for loading txt and pdf to filehandler.

nb currently blocked by - https://github.com/mindsdb/mindsdb-frontend/issues/1733

**Fixes** #6827 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 1]
```diff
@@ -101,29 +108,43 @@ def _handle_source(file_path, clean_rows=True, custom_parser=None):
...
+            docs = loader.load()
+            df = pd.DataFrame([{"text": doc.page_content} for doc in docs])
+
+        elif fmt == "pdf":
+            from langchain.document_loaders import PDFMinerLoader
+
+            loader = PDFMinerLoader(file_path)
+            docs = loader.load()
+            df = pd.DataFrame([{"text": doc.page_content} for doc in docs])
+
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:139]: additions here

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 2]
```diff
@@ -192,76 +216,84 @@ def _get_data_io(file_path):
...
+        if suffix == "xlsx":
+            return data, "xlsx", dialect
 
-        if suffix == 'xlsx':
-            return data, 'xlsx', dialect
+        if suffix == "txt":
+            return data, "txt", dialect
+
+        if suffix == "pdf":
+            return data, "pdf", dialect
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:246]: additions also here

**Comment by dusvyat** [comment]
I commented in points where I made actual changes, the rest of the changes are as a result of linting

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 3]
```diff
@@ -101,29 +108,43 @@ def _handle_source(file_path, clean_rows=True, custom_parser=None):
...
             df = pd.read_excel(data)
 
-        elif fmt == 'json':
+        elif fmt == "json":
             data.seek(0)
             json_doc = json.loads(data.read())
             df = pd.json_normalize(json_doc, max_level=0)
 
+        elif fmt == "txt":
+            from langchain.document_loaders import TextLoader
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:127]: if file handler is our base handler we need to add langchain to mindsdb requirements

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 4]
```diff
@@ -101,29 +108,43 @@ def _handle_source(file_path, clean_rows=True, custom_parser=None):
...
+
+            loader = TextLoader(file_path, encoding="utf8")
+            docs = loader.load()
+            df = pd.DataFrame([{"text": doc.page_content} for doc in docs])
+
+        elif fmt == "pdf":
+            from langchain.document_loaders import PDFMinerLoader
+
+            loader = PDFMinerLoader(file_path)
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:136]: this line raised 
ImportError: `pdfminer` package not found, please install it with `pip install pdfminer.six`
we need to add it to requirements too

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 5]
```diff
@@ -101,29 +108,43 @@ def _handle_source(file_path, clean_rows=True, custom_parser=None):
...
+            docs = loader.load()
+            df = pd.DataFrame([{"text": doc.page_content} for doc in docs])
```
Comment (ea-rus) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:138]: for pdf and text I've got dataframe that contents one row and one column with all content of pdf or text
Is it expected behavior?

Comment (dusvyat) [reply mindsdb/integrations/handlers/file_handler/file_handler.py:138]: no It shouldn't I will look into this

File: mindsdb/integrations/handlers/file_handler/file_handler.py (hunk 4 above)
Comment (dusvyat) [reply mindsdb/integrations/handlers/file_handler/file_handler.py]: fixed

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
-    HandlerStatusResponse as StatusResponse,
-    HandlerResponse as Response,
-    RESPONSE_TYPE
-)
+from mindsdb.integrations.libs.response import RESPONSE_TYPE
+from mindsdb.integrations.libs.response import HandlerResponse as Response
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+
+DEFAULT_CHUNK_SIZE = 200
+DEFAULT_CHUNK_OVERLAP = 50
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:27]: add default chunk size and overlap for txt and pdf

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 7]
```diff
@@ -36,14 +36,24 @@ class FileHandler(DatabaseHandler):
...
         super().__init__(name)
         self.parser = parse_sql
         self.fs_store = file_storage
-        self.custom_parser = connection_data.get('custom_parser', None)
-        self.clean_rows = connection_data.get('clean_rows', True)
+        self.custom_parser = connection_data.get("custom_parser", None)
+        self.clean_rows = connection_data.get("clean_rows", True)
+        self.chunk_size = connection_data.get("chunk_size", DEFAULT_CHUNK_SIZE)
+        self.chunk_overlap = connection_data.get("chunk_overlap", DEFAULT_CHUNK_OVERLAP)
         self.file_controller = file_controller
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:57]: chunk size and overlap can be passed in via` connection_data`

File: mindsdb/integrations/handlers/file_handler/file_handler.py [hunk 8]
```diff
@@ -101,29 +126,50 @@ def _handle_source(file_path, clean_rows=True, custom_parser=None):
...
             data.seek(0)
             json_doc = json.loads(data.read())
             df = pd.json_normalize(json_doc, max_level=0)
 
+        elif fmt == "txt" or fmt == "pdf":
+            from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+            text_splitter = RecursiveCharacterTextSplitter(
+                chunk_size=chunk_size, chunk_overlap=chunk_overlap
+            )
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/file_handler/file_handler.py:149]: this splits text by fixed character size

**Review by ea-rus — APPROVED** [review]

**Pull request merged**